### ISSUE TITLE
Top level await with SWC

### ISSUE THREAD

**Issue by timneutkens** [vercel/next.js#31054]

**Comment by marklawlor** [comment]
@glasser NextJS 12 is switching from Babel to SWC. Top level await does not appear to be configurable under SWC, so if you update to Next 12 you need to opt out of it (continue using Babel).

Whether this is actually blocked is a call for the Vercel team, but I assume they would prefer examples to use SWC unless they rely on a specific Babel plugin.

**Comment by sajadghawami** [comment]
in a previous comment i've read that `.swcrc` was added in `nextjs@12` but removed in `nextjs@12.1` due to certain issues.

if this was true and if this would be of any help, I would be glad to test `topLevelAwait` in my current application with `nextjs@12`.

**Comment by ctjlewis** [comment]
This is an incredibly important language feature and IMO it has been a mistake to deprioritize it in general and leave it as an experimental opt-in.

Top-level await is one of the only hangups that prevent valid JS programs from being executed inside Next, the others being:

- Next server context isolation (#29565)
- Experimental ESM externals

This is currently causing issues trying to import perfectly valid modules that contain top-level await statements, even with the flag enabled (as you mentioned, it's not well-tested).

**Comment by djvex** [comment]
Yeah I am new to developing with next.js and running into this issue as well.

I'm new to `next.js` and the JavaScript ecosystem in general.

In my `node_modules\next\dist\build\webpack\loaders\next-swc-loader.js` file I tried adding `exports.experiements.futureDefaults = true`. Is even getting my app to compile even possible or am I just stuck waiting for you guys to make an update at this point?
```
`"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.pitch = pitch;
exports.default = swcLoader;
exports.raw = void 0;
exports.experiments.futureDefaults = true; // I think I need this?
var _swc = require("../../swc");
var _options = require("../../swc/options");
var _path = require("path");
async function loaderTransform(parentTrace, source, inputSourceMap) {
    // Make the loader async
    const filename = this.resourcePath;
    let loaderOptions = this.getOptions() || {
```

**Comment by gabrieli** [comment]
Is there any workaround for using top level await with swc until the feature is supported?
Thanks!

**Comment by eric-burel** [comment]
This seems to prevent Apollo Server setup or at least to make it harder, the open PR to update to Apollo 3 also use a top level await: https://github.com/vercel/next.js/pull/30082/

**Comment by lodmfjord** [comment]
Any news on this?

**Comment by trevex** [comment]
I just ran into this issue and worked around it by using babel instead.

Just create a `.babelrc`, also see [docs](https://nextjs.org/docs/advanced-features/customizing-babel-config) and babel will be used instead.

**Comment by TommySorensen** [comment]
There is a `topLevelAwait` setting in the SWC documentation, does this mean that it's supported in SWC and therefore nextjs? https://swc.rs/docs/configuration/compilation
